### DISCUSSION TITLE
Give Heading UI element an instructions/description field

### DISCUSSION THREAD

**Discussion by RyanRoberts** [craftcms/cms#13513] [Ideas] [answered]
I've repeatedly found myself go into the Heading UI Element field options thinking it would have a field for a description. 

What would be good to have…
![Google Chrome - 2023-08-07--10-27-28@2x](https://github.com/craftcms/cms/assets/108874/89a5f628-aa2b-40e9-a6de-8384938b508b)

What I'm doing at the moment…
<img width="659" alt="Google Chrome - 2023-08-07--10-26-06@2x" src="https://github.com/craftcms/cms/assets/108874/dce902dd-048c-4ff3-b623-09b469fa2c0f">

Just an idea.

**Comment by MoritzLost** [comment]
+1 for this feature request. The tip UI element is nice, but very bright and attention-grabbing, which I don't want all the time. Being able to add simple instructions to a heading would be great:

![mockup-heading-with-instructions](https://github.com/user-attachments/assets/75f16769-ddda-4006-b310-55cdb763b3fa)

**Comment by brianjhanson** [comment, marked as answer]
We introduced a "markdown" UI element in 5.5 (https://github.com/craftcms/cms/pull/15674, https://github.com/craftcms/cms/discussions/15664) which I think is a better solution than adding a description to the heading field.

Is there a use case not covered by the markdown UI element?

**Reply by MoritzLost** [reply]
@brianjhanson Good point, that works 🎉
… to 99 % 🙂

Now I'm using a heading UI element followed by a markdown element. Already a good improvement over using a tip:

![Screenshot 2025-03-27 at 13 03 01](https://github.com/user-attachments/assets/3396e56f-6429-488b-819e-0a220d28ea49)

The only thing that bugs me is that the distance between heading and paragraph is a bit large, the paragraph is even closer to the next line of fields. I would prefer if the instructions were directly below the heading, similar to field headings, to indicate that the instructions belong to the heading / section. So it would still be a useful addition to have instructions for headings that are displaying directly below the heading.

**Reply by brianjhanson** [reply]
Yeah, it's a little larger than you'd want it to be. I just opened #16973 to adjust that spacing a bit if you throw both the heading and the description into the markdown UI element. The existing headings are `h2`s if you want the sizing to be equivalent.

**Reply by brandonkelly** [reply]
Craft 5.6.14 is out with the changes in #16973. Thanks for pointing that out!

**Reply by MoritzLost** [reply]
@brianjhanson @brandonkelly Thanks, that looks better!

Though it's still a bit difficult to achieve a consistent look and feel for headings/form sections with instructions. Here's a comparison:

![Screenshot 2025-04-08 at 13 34 57](https://github.com/user-attachments/assets/57a18f94-e461-4525-bbba-b5bcf8135597)

The first group of fields uses a Markdown UI element to provide the divider, heading and instruction text. Using just Markdown doesn't look great - the `<hr>` divider doesn't span the whole width and the heading uses a grey color, so it's inconsistent with regular heading UI elements, and also doesn't divide the form sections as clearly.

The second group uses a heading UI element follow by a Markdown UI element for the instructions. Here I have the spacing problem again.

So in summary, I would _still_ like to see an option to add instructions to a heading UI element 🙂

**Reply by brandonkelly** [reply]
The lighter text color wasn’t intentional, so just fixed that for the next release (1a3592b252ce51609e492e1c2d320efbf21cb122).

As for the HR, you can use a dedicated Horizontal Rule UI element in combination with the Markdown element if you want it to span the full width of the pane.

**Reply by MoritzLost** [reply]
@brandonkelly Got it, thanks! I'll give it another try after the next release 👍

**Reply by brandonkelly** [reply]
Craft 5.6.16 is out with that fix 🎉

**Reply by MoritzLost** [reply]
@brandonkelly Awesome, thanks! Combination of Horizontal Rule and Markdown with heading + paragraph now looks great and consistent 🎉
